### PR TITLE
[Translation] Fixed bug while editing taxon permalink

### PR DESCRIPTION
**Pull request by Zales0123**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Fixed tickets | https://github.com/Sylius/Sylius/issues/1624
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: src/Sylius/Bundle/TranslationBundle/GedmoHandler/TranslationSlugHandler.php [hunk 1]
```diff
@@ -128,6 +128,8 @@ public function postSlugBuild(SluggableAdapter $ea, array &$config, $object, &$s
                 }
             }
         }
+
+        $slug = $this->deleteUnnecessaryParentSlug($slug);
```
Comment (stloyd) [inline src/Sylius/Bundle/TranslationBundle/GedmoHandler/TranslationSlugHandler.php]: This param is never used.

Comment (stloyd) [inline src/Sylius/Bundle/TranslationBundle/GedmoHandler/TranslationSlugHandler.php]: I don't see point for placing new one-line method that runs only preg replace.

Comment (pjedrzejewski) [inline src/Sylius/Bundle/TranslationBundle/GedmoHandler/TranslationSlugHandler.php]: Yeah, for this one function call I think it could stay here.

Comment (Zales0123) [inline src/Sylius/Bundle/TranslationBundle/GedmoHandler/TranslationSlugHandler.php]: *Primo:* Unfortunately ``$slug`` is passed by reference, so it's in fact used somewhere else ;)
*Secundo:* The reason is simple - clean code principles. Let's assume we have that line in the code:
```php
$slug = preg_replace('/^' . preg_quote($this->parentSlug.$this->usedPathSeparator, '/') . '/', '', $slug);
```
Even if your fluent in regular expressions, it takes a to figure out what does this line do. I would probably need a comment to explain it. Seeing this:
```php
$slug = $this->deleteUnnecessaryParentSlug($slug);
```
we are 100% sure what the hell is going on. Just following Uncle Bob advices ;)

Comment (pjedrzejewski) [inline src/Sylius/Bundle/TranslationBundle/GedmoHandler/TranslationSlugHandler.php]: Where is Secundo? :D I agree. Let's keep it this way.

Comment (pjedrzejewski) [inline src/Sylius/Bundle/TranslationBundle/GedmoHandler/TranslationSlugHandler.php]: Ok I see now. Need to wake up. :)

**Comment by nakashu** [comment]
@pjedrzejewski Why this wasn't merged ? It will also fix this patialy #3662, 3rd example.
@Zales0123 - can you rebase it or should I?

**Comment by Zales0123** [comment]
@nakashu my pleasure with rebasing. I suppose they are too many PR's and such a small change could be easily lost. Nevertheless, @pjedrzejewski if you think it's good to merge - it's ready ;)

**Comment by michalmarcinkowski** [comment]
Thank you Mateusz!

**Pull request merged**